### DISCUSSION TITLE
Control Bar fixes and improvements

### DISCUSSION THREAD

**Discussion by xezon** [TheSuperHackers/GeneralsGamePatch#1543] [General]
Topic for all things Control Bar.

TBD.

**Comment by xezon** [comment]
I am in favor of changing Overlord Upgrade icons in same style as Helix Upgrade icon. Reason being that the Helix Upgrade icons are much more readable than the Overlord counterpart. The blue triangle with the structure representation does a good job of showing what it gives. The Overlord upgrade icons are a bit more difficult to look at because of their small representation of the upgrade.

We can reuse the art of the original Overlord Upgrade icons. They have a lot of dead space (the sky) which can fit the blue triangle of the Helix Upgrade icons.

![shot_20230111_221901_2](https://user-images.githubusercontent.com/4720891/211921829-0410a119-89c0-4122-b377-3516bf5d69e0.jpg)

![shot_20230111_221918_4](https://user-images.githubusercontent.com/4720891/211921830-e35d35e7-9b6c-40eb-bcfe-86e17f858276.jpg)

**Comment by ExiLe-ZH** [comment]
I can make that

**Comment by ExiLe-ZH** [comment]
![overlord upgrades](https://user-images.githubusercontent.com/32326277/211948377-d76b58a2-295c-4051-943a-5e2b934ad894.png)

Probably need to zoom in bit more on the overlord, same like the lix icon. The background is also too white? Needs to be more grey? Or need a better overlord source picture all together?

**Reply by MTKing4** [reply]
This one is good, but white background needs to be darker yes,

**Reply by xezon** [reply]
We can also change the blues to reds and oranges to fit the China color theme better.

**Comment by ExiLe-ZH** [comment]
![overlord upgrades v2](https://user-images.githubusercontent.com/32326277/211949329-72f6a0a5-aede-4a16-aa3f-73d37f263712.png)

better?

**Comment by Stubbjax** [comment]
This suggestion feels like a regression to me. I personally hate the Helix's upgrade icons in 1.04 - they are sloppy compositions of other structures / upgrades and clearly the result of a quick copy / paste job and tight deadlines. The saturated blue also looks terrible and conflicts with the rest of the UI. They look much better in mods like ShockWave, which features dedicated portraits for each upgrade.

![image](https://user-images.githubusercontent.com/11547761/211968305-ce094c88-0a9c-4be8-9f74-2d305695f097.png)

It would also be a regression to apply this button style to USA's vehicle drones if consistency was to be pursued. It would be far better if dedicated Helix upgrade portraits were created instead; then this cut-in-half button style could exclusively represent fake structures for a much cleaner design.

**Reply by xezon** [reply]
This is a reasonable objection. I like the blue triangle icons for their clarity, but we can also go different route. As for the Shockwave Icons, they are okay too. The backdrop should be brighter to show more contrast. This can be achieved by using snowy terrain, which would be consistent with other Helix icons.

**Reply by MTKing4** [reply]
I agree with stubjax here, but I don't really prefer the shockwave icons

**Comment by MTKing4** [comment]
I think the best thing to do is to leave things as is, i like current overlord icons as they are. And yes they do look better than helix's

**Reply by xezon** [reply]
I disagree with this. I do remember, when I was a beginner player, it took me a very long time to recognize the difference between the Overlord Propaganda and Overlord Bunker Icon. That is because they look very similar.

**Reply by MTKing4** [reply]
I've never had such issues myself nor have i heard it from anyone else besides you, i think it's a matter of taste, and I don't think the helix design choice is an improvement, that being said I don't mind seeing something with higher quality for ovies but similar to the current design

**Reply by Stubbjax** [reply]
> I disagree with this. I do remember, when I was a beginner player, it took me a very long time to recognize the difference between the Overlord Propaganda and Overlord Bunker Icon. That is because they look very similar.

Your personal experience as a new player is not enough justification for changing something for everyone 20 years later. Nobody is a beginner player anymore and everybody is familiar with the current buttons. Whether the current buttons are confusing is very questionable. Changing this now would only cause more confusion.

**Reply by ExiLe-ZH** [reply]
Tbh the original pictures are more than clear enough, you exactly can see the upgrade models. Moreover you have the information windows with upgrade names and descriptions.

I do agree though that there's a lot of wasted background space in the icons, which I don't personally like either, neither the colors of it.

**Reply by xezon** [reply]
> Your personal experience as a new player is not enough justification for changing something for everyone 20 years later. Nobody is a beginner player anymore and everybody is familiar with the current buttons. Whether the current buttons are confusing is very questionable. Changing this now would only cause more confusion.

I do trust my judgement on this issue. I can back it up with numbers too. To illustrate problem we can compare cameo images overlap. To optimize the test, I cleaned up minor artifacts in the icons, such as the blurry Helix on the Helix Propaganda icon. Then I used online tool https://www.imgonline.com.ua/eng/similarity-percent.php to calculate similarity.

Training vs Chainguns yields **43.64 %** similarity. It is fair to say these icons look different.

![training](https://user-images.githubusercontent.com/4720891/212314885-6d52203f-ab2d-482a-a8f1-a12bf819cb64.jpg)
![chainguns](https://user-images.githubusercontent.com/4720891/212312518-cd98171b-4482-4a81-91a4-f54e43845817.jpg)

Helix Bunker vs Helix Gattling yields **82.18 %** similarity.

![helix_bunk](https://user-images.githubusercontent.com/4720891/212313093-d3bc0f6c-d059-428c-a9e7-0522c3e7f7cf.png)
![helix_gatt](https://user-images.githubusercontent.com/4720891/212313108-46d60f2d-aeeb-4a84-9863-fe64f6f8eab2.png)

Helix Bunker vs Helix Propaganda yields **83.21 %** similarity.

![helix_bunk](https://user-images.githubusercontent.com/4720891/212313093-d3bc0f6c-d059-428c-a9e7-0522c3e7f7cf.png)
![helix_prop](https://user-images.githubusercontent.com/4720891/212313184-11d5c928-017f-47a9-90e6-3a1689e3eb60.png)

Overlord Bunker vs Overlord Gattling yields **95.07 %** similarity.

![ovr_bunk](https://user-images.githubusercontent.com/4720891/212313280-745ea944-aa33-4e16-bac1-b98f1e54ae7f.png)
![ovr_gatt](https://user-images.githubusercontent.com/4720891/212313288-685e614d-d124-4b6d-ba9e-70e5c6d533ff.png)

Overlord Bunker vs Overlord Propaganda yields **95.95 %** similarity.

![ovr_bunk](https://user-images.githubusercontent.com/4720891/212313280-745ea944-aa33-4e16-bac1-b98f1e54ae7f.png)
![ovr_prop](https://user-images.githubusercontent.com/4720891/212313322-d98b7924-0088-4ccd-ac3e-a537b156c87c.png)

As numbers show, Overlord icons show high percentage of similarity. Much more than the Helix icons. Overlord Bunker and Overlord Propaganda just have a 4% difference.

I understand you want to not to change it because you are used to it, but factually these Overlord icons look way too similar and it would be an improvement to make them more distinguishable.

Here are the uncompressed icons used for comparison: [cameos2.zip](https://github.com/TheSuperHackers/GeneralsGamePatch/files/10411228/cameos2.zip)

**Reply by ExiLe-ZH** [reply]
Humans process such information much different than machines. I honestly really don't know how you can be confused when you see all 3 pictures in front of you, your attention should be drawn to the very notable differences there are.

Pictures should even be more recognizable after having seen the models ingame once, it's exactly advertized like so.

**Reply by Stubbjax** [reply]
> As numbers show, Overlord icons show high percentage of similarity. Much more than the Helix icons. Overlord Bunker and Overlord Propaganda just have a 4% difference.
>
> I understand you want to not to change it because you are used to it, but factually these Overlord icons look way too similar and it would be an improvement to make them more distinguishable.

And yet to me, the overlord upgrades are visually simpler and easier to read.

These comparisons just come across as pseudoscience. You can have 0.5% similarity between two images and still easily detect the difference.

![image](https://user-images.githubusercontent.com/11547761/212319920-0d4d0d95-452b-40d6-8322-464bb0eba574.png) ![image](https://user-images.githubusercontent.com/11547761/212319901-6c5da4f1-7a45-492f-8894-4fee3c5e7479.png)

And you can have two images that look identical and have 0% similarity.

![1](https://user-images.githubusercontent.com/11547761/212319750-70aa7983-9dfb-436c-afe0-b6ecee77401f.png) ![2](https://user-images.githubusercontent.com/11547761/212319754-79260ffd-0ed2-41da-a9f8-8c46d81902e0.png)

Humans do not work like machines.

**Reply by xezon** [reply]
The red dot image is a fair comparison to the actual problem. The dot is small, as are the upgrades on the icons. If the depiction of the red dot was larger, then the observable difference would correlate and become larger. Greater image difference can be achieved by giving the difference more space, contrast or color.

Calling it pseudo science is arrogant.

**Reply by Stubbjax** [reply]
It merely comes across that way as you are trying to use mathematics to explain psychology.

**Reply by ExiLe-ZH** [reply]
> The red dot image is a fair comparison to the actual problem. The dot is small, as are the upgrades on the icons. If the depiction of the red dot was larger, then the observable difference would correlate and become larger.

That's not the point though, the small difference stands out for us, because the rest is all exactly the same.

**Reply by xezon** [reply]
> Humans process such information much different than machines. I honestly really don't know how you can be confused when you see all 3 pictures in front of you, your attention should be drawn to the very notable differences there are.

There is a strong correlation between the measurable difference between 2 images and the human recognizable difference. The Overlord icons have small differences as illustrated above. You are able to see those differences when you look closely. If the differences were drawn larger and more clearly, then it would also be easier for a human to notice these differences.

To re-illustrate the problem, imagine we zoom out this picture further and show the entire Overlord model plus plenty sky. The Overlord attachment just covers a few pixels. Would you be confident to say that the visibility of the difference is unchanged?

**Reply by MTKing4** [reply]
I kinda agree with stubjax and Exile, humans see things differently than machines as human eyes have a small focus area and tend to look at the difference not the similarity, we should approach this with psychology and human biology over mathematics,

As you can see in your first example, mathematics say there's 43% similarity, whereas there's zero similarity to the human eyes, so mathematics are not to be trusted here.

> Training vs Chainguns yields **43.64 %** similarity. It is fair to say these icons look different.

![training](https://user-images.githubusercontent.com/4720891/212314885-6d52203f-ab2d-482a-a8f1-a12bf819cb64.jpg)
![chainguns](https://user-images.githubusercontent.com/4720891/212312518-cd98171b-4482-4a81-91a4-f54e43845817.jpg)

**Reply by xezon** [reply]
43 % similarity is a reasonable value for this comparison, because only the difference of black to white should yield 0%. It simply measures difference of pixel color values. It is a simple and unbiased way to compare.

**Reply by MTKing4** [reply]
I know it measures similarity by pixels and their color range and location on the image array, but we humans do not look at pixels, this is not a relevant comparison for us because in the end it's players who are going to play the game not bots, and players will see no similarity here

**Reply by xezon** [reply]
Yes, measurements can be different from what humans perceive. For example you can have a house which looks normal. Most visitors may not notice anything unusual. But a measurement may reveal that the house is not straight and leans a few degrees to one side. This pitch could indicate a problem with the house. You can choose to ignore the measurement, or consider it. Personally, I would be cautious to buy a house that should be straight but is not straight, even if I do not perceive it that way by just looking at it.

If you choose to dismiss the measurements, then so be it. I will not argue about it. It just means we cannot have a factual discussion, as you will just focus on opinion.

**Reply by MTKing4** [reply]
You can't compare a house to a picture, a house is not just meant to be looked at, it's meant to be lived and and moved around within which will have consequences but also things that humans will perceive that the house angle is pitched, like the direction the water goes when spilled on the ground, 

With pictures all we do is look at it, and if it looks good and well represented, there's nothing needed to be done about it.

**Reply by xezon** [reply]
It depends on what measurements you decide to dismiss. You may not dismiss house measurements, but you chose to dismiss the icon comparison measurements. Since you ignore facts and solely side with your opinion(s), we cannot discuss this properly.

**Reply by ExiLe-ZH** [reply]
Hard to have a factual discussion while the result is all perception.

**Reply by MTKing4** [reply]
I think you don't seem to understand the fact that art Can't be measured, this is not an equation or a code for you to measure, a lot of art is about taste and opinion and how it looks to the perceiving eyes, i think the best people to decide on this matter are people with art experience

**Reply by ExiLe-ZH** [reply]
> To re-illustrate the problem, imagine we zoom out this picture further and show the entire Overlord model plus plenty sky. The Overlord attachment just covers a few pixels. Would you be confident to say that the visibility of the difference is unchanged?

That's the thing, when I see those 3 pictures next to each other, the small differences/shapes draw my attention instantly. My brain filters the useless information automatically. 

One could however have eyesight problems and not find the difference at all, that's a legit complaint.

**Reply by xezon** [reply]
It is ok, I can tackle this on my own then.

**Comment by Stubbjax** [comment]
Here is an example of how things might look if consistency was pursued without regard for minimizing the differences with 1.04 command sets.

All Evacuate commands are now located in the same position and visually identical, allowing all transport units to be evacuated in grouped selections. Overlord and Helix attachment upgrades match the USA drone upgrade positions / order as it is the most common.

![image](https://user-images.githubusercontent.com/11547761/212465506-77601333-b9aa-4c72-aca3-84ed3275b7a0.png)
Click [here](https://user-images.githubusercontent.com/11547761/212465578-2dcc3c5c-3b06-474e-aadb-7bb6b6d322fb.gif) for an animated view.

The differences of each row are listed below. Upgrade cameos are removed to nullify irrelevant differences. Row references are denoted by `[#]` in the descriptions below. Note that several of these differences already exist in the 1.04p main branch, which are marked as `[1.04p]`.

## 1: Overlord attachment upgrade layout

The command set layout of the standard Overlord has been altered to match the USA's standard vehicle drone upgrades layout `[4]`. This is cleaner than the [layout](https://user-images.githubusercontent.com/4720891/212160668-9c479f99-be4f-435f-b8a7-c51c66b14ab2.jpg) suggested in #1544. (There is even a strangely satisfying symmetry to the command set. A case could be made for achieving a higher level of consistency with the 'corner' layout of the Attack Move, Guard and Stop actions.)

The attachment upgrades are ordered in a way that matches the USA drone types:

Slot 1 = Gattling Cannon / Battle Drone (gun)
Slot 2 = Speaker Tower / Scout Drone (non-combat)
Slot 3 = Bunker / Hellfire Drone (effectively rockets)

## 2: Overlord transport slot and Evacuate layout

The Overlord's transport slots and Evacuate button `[2]` now match the USA Humvee `[4]`, which is the transport command set layout that players are most familiar with. Evacuate functionality will now work with both units grouped in the same selection.

## 3: Emperor attachment upgrade layout

As a result of changing the standard Overlord's upgrade layout to match USA's drone upgrade layout `[1]`, the Emperor Overlord's Gattling Cannon attachment upgrade can be moved from slot 3 to slot 1, which feels nicer.

## 4: Humvee reference

The Humvee's Evacuate button has been standardised, but has otherwise undergone no changes and is primarily included for reference.

## 5: Helix attachment upgrade, transport slot and Evacuate layout

The command set layout of the standard Helix has been shifted to match the USA's standard vehicle drone upgrades layout `[4]` and new Overlord attachment upgrade layout `[1]`.

As a result, the transport slots and Evacuate button are adjusted to match the Humvee `[4]`. Furthermore, the Napalm Bomb upgrade and ability have been moved to the remaining available slots 10 and 12, respectively.

## 6: Upgraded Helix layout

The upgraded Helix layout serves as a reference to show how the Helix's upgraded state differs from 1.04.

## 7: Infantry Helix Napalm Bomb upgrade and Evacuate layout

The Evacuate button has been swapped with the Napalm Bomb so that it is standardised with other units. Evacuate functionality will now work with other transport units grouped in the same selection. The Infantry Helix's Napalm Bomb upgrade / ability consistency (or lack thereof) from 1.04 is preserved.

## 8: Ambulance Evacuate and Clear Toxins layout

The Evacuate button of the Ambulance now matches the Humvee, allowing evacuation when both are grouped in the same selection `[1.04p]`. The Clear Toxins ability is also shifted from slot 10 to 12 to align with other unit abilities, such as the Helix's Napalm Bomb and Demolition General's Suicide ability.

## 9: Chinook Evacuate button and Combat Drop ability layout

The Evacuate button of the Chinook has been standardised with the Evacuate buttons of ther units, which improves clarity (these two buttons can be difficult to distinguish) and allows Evacuate functionality to work with other transport units grouped in the same selection.

As with the Ambulance `[8]`, the Combat Drop ability is shifted from slot 10 to 12 to align with other unit abilities. The button border colour is also green instead of yellow, as it is a special ability.

## 10: Listening Outpost Evacuate layout

The Evacuate button now matches other transport units. Evacuate functionality will now work with other transport units grouped in the same selection `[1.04p]`.

## 11: Troop Crawler reference

A different change has given the Troop Crawler a Guard action `[1.04p]`, but it has otherwise undergone no changes and is included for reference.

## 12: Battle Bus Evacuate layout

The Evacuate button now matches other transport units. Evacuate functionality will now work with other transport units grouped in the same selection `[1.04p]`.

## 13: Technical Evacuate layout

The Evacuate button now matches other transport units. Evacuate functionality will now work with other transport units grouped in the same selection `[1.04p]`.

## Further comments

Despite the changes, there are still consistency issues. Non-transport units and all infantry have their special abilities located in the top-left slots as opposed to the bottom-right slots. There is no way to standardise the positions of special abilities without even more radical changes than this one and introducing even greater usability issues. It is quite likely impossible.

The Combat Cycle's Evacuate button remains in slot 1 due to the irreversible nature of the action, as well as the extreme unlikelihood that a player would want the action to occur while evacuating other grouped transport units. It is technically not even a true transport unit as it lacks any transport slots.

**Reply by xezon** [reply]
I applaud your effort to create this fantastic overview. Could not be any clearer.

### Regarding 1: Overlord attachment upgrade layout

See next passage.

### Regarding 2: Overlord transport slot and Evacuate layout

> now match the USA Humvee, which is the transport command set layout that players are most familiar with

I do not agree with this statement. USA Humvee and USA Ambulance Command Sets are outliers. I skimmed over all Command Sets and every other vehicle and structure has Command_TransportExit or Command_StructureExit begin at button index 1. This means the most common exit buttons are not placed like they are in USA Humvee, Ambulance. Now this may be perceived subjectively if a player plays USA faction only and builds Humvees all the time, but even then, Chinooks, Firebase and all Civilian buildings have exit buttons starting at index 1. Surely USA player interacts with those too.

Therefore, when attempting to achieve consistency among command set setups, then it would actually make more sense to move Humvee, Ambulance transport exit slots to index 1. Because then all units and structures would follow identical rule for exit slots.

I do not think it is necessary to touch Humvee, Ambulance command sets for our purposes, but adhering to index 1 position for Overlord exit slots looks more reasonable to me than adhering to Humvee command sets, because as explained above, the index 1 concept is much more common across all units and structures.

What I can say however is that the Humvee layout makes sense for Overlord for 3 reasons:

1. Evacuate button will be closer to transport exit slots
2. Emperor Gatling Upgrade button can sit alone at sweet index 1
3. Buttons are pushed nicely to the side which presents no (temporary) gaps to the left of it

It is unclear to me if these 3 reasons are prime to justify it. Because the disadvantages are:

1. Inconsistency with index 1 exit slots of Bunker, Outpost, Troopcrawler, etc
2. Inconsistency of Bunker Upgrade position with Infantry Helix at least, preventing shortcut
3. Requires Upgrade Button reposition on both Helix and Overlord
4. Requires Transport Slot Button reposition on both Helix and Overlord

To me it looks like the Humvee layout provides some satisfying visuals, at the expense of functionality, button consistency and increased difference to Original layouts.

### Regarding 7: Infantry Helix Napalm Bomb upgrade and Evacuate layout

It would be good if Infantry Helix Bunker Upgrade button and other Helix Bunker Upgrade buttons are on same position. This would allow the use of button in group. With the new proposal, this functionality is not given and the Helix Bunker Upgrade buttons are far apart. The original Helix Upgrade button positions would allow to do this, by swapping Helix Upgrade positions with each other.

> The Evacuate button has been swapped with the Napalm Bomb so that it is standardised with other units. Evacuate functionality will now work with other transport units grouped in the same selection.

I was thinking about Helicopter Evacuation buttons and did not take issue with them being placed differently than ground vehicles. Reason being, it is not clear to me if vehicle + helicopter in combined group selection should provide evacuation functionality. There is a case to make that it may be generally undesired to evacuate a ground vehicle and air vehicle in one go. What scenario would favor a grouped evacuation? And what scenario would not favor this? However, on the other hand this decision can be made in isolation from Evacuate button position, simply by using identical or 2 different Command buttons for ground and air vehicles, that either allow or block grouped evacuation.

It certainly looks desirable however to have Evacuate button for Chinooks and Helix on same position, because Chinooks and Helix drops are conceptually identical.

**Comment by ExiLe-ZH** [comment]
Two new style proposals for Overlord/Lix Upgrade Buttons:

Proposal 1 (Original style in red):
![old style old bar](https://user-images.githubusercontent.com/32326277/213490442-c4297b58-341f-4f24-94d0-33db514ab950.png)
![overlord old style old bar](https://user-images.githubusercontent.com/32326277/213493377-514c3fc9-3c9b-44ce-ba93-63a3a5513f87.png)

Proposal 2 (New style):
![lix new buttons](https://user-images.githubusercontent.com/32326277/213486223-f595d1ea-edcb-45ee-abfc-251141e53b76.png)
![overlord new style old bar](https://user-images.githubusercontent.com/32326277/213496072-e738308e-6b29-44c9-aa51-61aa80021218.png)

**Comment by xezon** [comment]
Eventually create new Cancel Build button image to be distinct from Unit Stop button as per discussion in #1572.

**Comment by ExiLe-ZH** [comment]
Rally point command icons (+hotkeys) on production buildings are redundant and take up controlbar space. I think it's safe to remove this feature?

**Reply by xezon** [reply]
Hmm good point. Does the button have any purpose?

**Reply by ExiLe-ZH** [reply]
Upon selection the cursor is already the waypoint type by default, it seems there's no purpose whatsoever?

**Reply by kABUSE1** [reply]
> Hmm good point. Does the button have any purpose?

Can confirm that you can still place a rally point even if you remove the command button "Command_SetRallyPoint" from an object type. So using it for something else shouldn't break anything.

**Reply by xezon** [reply]
Yeah functionality wise it is obsolete. The only information this gives is that a structure can have a rally point. But this information is also given by the cursor icon. I think we can remove it.

**Reply by xezon** [reply]
Addressed with #1808

**Comment by kABUSE1** [comment]
Just as a side note, if you guys are out of space: there are four additional command button slots (15+16+17+18) that fully work. You'd just have to make them visible per command bar UI mod. 18 is the hard limit, anything higher will crash the game.

Correction: The hard limit seems to be 18, not 16.

**Reply by xezon** [reply]
I hope and think that this will not be necessary, as this would require redesign of original Control Bar and would also significantly change the look of it.

**Comment by ExiLe-ZH** [comment]
The Nuke BM has a nuclear icon in it's main cameo visible when the unit is selected. 

![image](https://user-images.githubusercontent.com/32326277/230611376-c72e865f-74a6-409a-a837-cfd1c3a34523.png)

This is missing from the Nuke Overlord, aswell as from both units in the WF. Should it be displayed in all cases? 

![image](https://user-images.githubusercontent.com/32326277/230611312-bb13ffba-7c51-46b2-bda3-cf8b3352b492.png)

**Reply by xezon** [reply]
I am not even sure I like this nuke icon at this point. commy2 saw it in NProject and copied it. But I think design is odd, because other icons show no symbols either, other than veterancy.

**Reply by kABUSE1** [reply]
> I am not even sure I like this nuke icon at this point. commy2 saw it in NProject and copied it. But I think design is odd, because other icons show no symbols either, other than veterancy.

This was most definitely done by EA because they were too lazzy to add a custom command button picture for nuke battle masters even though they use a different model.

The nuke sign should be removed and instead a new command button picture for nuke battle masters should be implemented. I'd consider that the clean way of fixing this. I can't think of other units having a custom model without also having their own custom command button picture.

**Reply by MTKing4** [reply]
I think it should be displayed everywhere, it's good to differentiate them from Non Nuke ones

**Reply by xezon** [reply]
> This was most definitely done by EA because they were too lazzy to add a custom command button picture for nuke battle masters even though they use a different model.

No this nuke symbol was added by NProject mod. Original game simply uses identical button art.

> The nuke sign should be removed and instead a new command button picture for nuke battle masters should be implemented. I'd consider that the clean way of fixing this. I can't think of other units having a custom model without also having their own custom command button picture.

This would be appropriate option yes.

**Reply by ExiLe-ZH** [reply]
> > This was most definitely done by EA because they were too lazzy to add a custom command button picture for nuke battle masters even though they use a different model.
> 
> No this nuke symbol was added by NProject mod. Original game simply uses identical button art.
> 
> > The nuke sign should be removed and instead a new command button picture for nuke battle masters should be implemented. I'd consider that the clean way of fixing this. I can't think of other units having a custom model without also having their own custom command button picture.
> 
> This would be appropriate option yes.

It's also in the HD control bar then.

**Reply by xezon** [reply]
The Control Bar HD uses original art only. No gimmicks.

Edit: Ah now I understand. The Portrait image has nuklear icon, but the button does not, correct?

**Reply by MTKing4** [reply]
Yep

**Reply by xezon** [reply]
Ok. Then the Nuke symbol on Button art is understandable.
